src/parser/constructs.rs

```diff
@@ -64,6 +65,11 @@ impl Construct {
         Ok((input, expr))
     }
 
+    /// Parse as many instructions as possible
+    pub fn many_instructions(input: &str) -> IResult<&str, Vec<Box<dyn Instruction>>> {
```

Comment:
This should return a ``ParseResult<...>``, no ?

Reply:
Yes, you're completely right!